project/src/UI/Upgrades/UpgradeButton.gd

```diff
@@ -3,23 +3,84 @@
 tool
 extends TextureButton
 
+signal on_ani_complete(is_entry)
+
 export var texture: Texture setget set_texture
 export var text := "" setget set_text
 
 onready var texture_rect := $VBoxContainer/TextureRect
 onready var label := $VBoxContainer/Label
 
+onready var tween := $Tween
+
+onready var on_pos = self.rect_position
+onready var off_pos = self.rect_position + Vector2(0, 200);
+
+onready var is_entry_ani = true;
+
+
+func animate(entry: bool, delay: float) -> void:
```

Comment:
For animations, I wouldn't recommend having a single function. The code is much harder to read than with two functions dedicated to each animation. Animations tend to be unique. It's also a bad example of code with seven conditional branches just to define the animation's parameters.